Task13

andreyv24 opened the pull request.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
.

andreyv24 committed:
Merge branch 'KFalcon2022:master' into master

andreyv24 committed:
Merge branch 'KFalcon2022:master' into task5

andreyv24 committed:
.

andreyv24 committed:
Merge remote-tracking branch 'origin/task5' into task5

# Conflicts:
#	src/main/java/com/walking/intensive/chapter1/task5/Task5.java

andreyv24 committed:
Merge branch 'KFalcon2022:master' into master

andreyv24 committed:
.

andreyv24 committed:
- Выделил проверки на некорректные данные в отдельный метод.
- Убрал оператор continue в цикле.
- Добавил пустые строки для читаемости.
-

andreyv24 committed:
Скорректировал на более короткую запись изменения значения переменной.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter1/task5/Task5.java:
```diff
@@ -125,4 +134,4 @@ static double getAreaAdvanced(double a, double b, double c) {
         return 0; // Заглушка. При реализации - удалить
     }
-}
+}
```
этого класса не должно быть в ПР

andreyv24 replied:
Это хвосты с Task5 тянутся в ПР других Task'ов. Устал уже удалять их.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,28 @@
+            counter++;
+            if (currentWateringCanVolume < plants[i]) {
+                counter += 2 * (i);
+                currentWateringCanVolume = wateringCanVolume - plants[i];
```
Достаточно оставить внутри if логику "наполнения" лейки и не придется городить огород с continue

andreyv24 replied:
Вроде убрал "огород" с continue.

KFalcon2022 commented on src/test/java/com/walking/intensive/chapter1/task5/Task5Test.java:
```diff
@@ -29,6 +29,7 @@ void getAreaByHeronTest() {
     void getHeightsTest() {
         assertArrayEquals(new double[]{4.615384615384615, 5.0, 12.0}, getHeights(12, 13, 5), 1e-10);
         assertArrayEquals(new double[]{2.4, 3.0, 4.0}, getHeights(3, 4, 5), 1e-10);
+        assertArrayEquals(new double[]{4.615, 5.0, 12.0}, getHeights(12, 13, 5), 1e-10);
```
Этого класса в ПР быть не должно

andreyv24 replied:
Удалил файлы не относящиеся к Task13.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,28 @@
+        int counter = 0;
+        int currentWateringCanVolume = wateringCanVolume;
+        for (int i = 0; i < plants.length; i++) {
+            if (plants[i] <= 0) {
```
То есть мы можем дойти до конца грядки и понять, что все вычисления были избыточны, потому что входные данные некорректны?)

Логичнее вынести валидацию в отдельный метод и проверять все до основной логики. Дополнительный проход по циклу чаще всего будет дешевле, чем выполнение массы вычислений вхолостую

andreyv24 replied:
> То есть мы можем дойти до конца грядки и понять, что все вычисления были избыточны, потому что входные  данные некорректны?)

Типа того. ))))

> Логичнее вынести валидацию в отдельный метод и проверять все до основной логики. Дополнительный           
проход по циклу чаще всего будет дешевле, чем выполнение массы вычислений вхолостую

Не задумывался об этом. Спасибо!

Вынес проверки в отдельный метод с прогоном по циклу на предмет неправильных значений.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,28 @@
+            }
+            currentWateringCanVolume = currentWateringCanVolume - plants[i];
+        }
+        return counter;
```
не стесняйся отделять логические блоки кода пустыми строками. Иначе весь метод сливается в одно полотно кода

andreyv24 replied:
Добавил пустые строки. Годится ?

andreyv24 committed:
Delete src/main/java/com/walking/intensive/chapter1/task5/Task5.java

andreyv24 committed:
Delete src/test/java/com/walking/intensive/chapter1/task5/Task5Test.java

andreyv24 committed:
Merge remote-tracking branch 'origin/task13' into task13

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,43 @@
     static int getStepsCount(int[] plants, int wateringCanVolume) {
-        // Ваш код
-        return 0;
+
```
лишняя пустая строка

andreyv24 replied:
Убрал строку.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,43 @@
+        return counter;
+    }
+
+    static boolean checkInputData(int[] plants, int wateringCanVolume) {
```
булеан-методы традиционно именуются с if

andreyv24 replied:
Ок. Поменял.

KFalcon2022 commented on src/main/java/com/walking/intensive/chapter3/task13/Task13.java:
```diff
@@ -49,11 +49,43 @@
+                counter += 2 * (i);
+                currentWateringCanVolume = wateringCanVolume - plants[i];
+            } else {
+                currentWateringCanVolume = currentWateringCanVolume - plants[i];
```
Можно упростить: `currentWateringCanVolume -= plants[i];`

andreyv24 replied:
Согласен. Исправлено!

KFalcon2022 commented:
На будущее: чтобы убрать файл из ПР его не нужно удалять. Нужно лишь привести его к виду, в котором он находится в целевой ветке. В твоем случае - в master

andreyv24 committed:
- удалил лишнуюю пустую строку.
- сократил выражение с уменьшением значения переменной.
- изменил название булевого метода.

andreyv24 committed:
- убрал пустую строку.
